src/main/java/mate/academy/dao/jdbc/UserDaoJdbcImpl.java

```diff
@@ -0,0 +1,174 @@
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't create user " + user.getLogin(), ex);
+        }
+    }
+
+    @Override
+    public Optional<User> getById(Long id) {
+        User user = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement("SELECT * FROM users WHERE user_id = ?"
+                            + " AND deleted = FALSE");
+            statement.setLong(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't getting user by id" + id, ex);
+        }
+        return Optional.ofNullable(user);
+    }
+
+    @Override
+    public Optional<User> findByLogin(String login) {
+        User user = null;
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement("SELECT * FROM users WHERE login = ?"
+                            + " AND deleted = FALSE");
+            statement.setString(1, login);
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                user = getUserFromResultSet(resultSet);
+            }
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't getting user by login" + login, ex);
+        }
+        return Optional.ofNullable(user);
+    }
+
+    @Override
+    public List<User> getAll() {
+        List<User> users = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement("SELECT * FROM users WHERE deleted = FALSE");
+            ResultSet resultSet = statement.executeQuery();
+            while (resultSet.next()) {
+                users.add(getUserFromResultSet(resultSet));
+            }
+            return users;
+        } catch (SQLException ex) {
+            throw new DataProcessingException("Couldn't getting all users", ex);
+        }
+    }
+
+    @Override
+    public User update(User user) {
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement statement = connection
+                    .prepareStatement("UPDATE users SET login = ?, password = ? "
+                            + "WHERE user_id = ? AND deleted = FALSE");
+            statement.setString(1, user.getLogin());
+            statement.setString(2, user.getPassword());
+            statement.setLong(3, user.getId());
+            statement.executeUpdate();
+            deleteFromRolesUsers(connection, user.getId());
+            insertIntoRolesUsers(connection, user);
```

Comment:
The method names are confusing. Maybe you could rename them as something like `deleteUserRoles()` and `addUserRoles()`?

Reply:
ok)